### DISCUSSION TITLE
Fail to build v0.5.0 in Windows 10: Cannot open include file: 'pthread.h'

### DISCUSSION THREAD

**Discussion by abribanez** [Autodesk/maya-usd#1001] [Q&A]
**Describe the issue**
Hi all!
I'm trying to build maya-usd for Maya 2020 but I keep getting `Cannot open include file: 'pthread.h'` in the logs and I haven't had any luck finding a solution in other tickets/issue-threads. Any help would be greatly appreciated. Cheers!

**Build log**
[build_log.txt](https://github.com/Autodesk/maya-usd/files/5685968/build_log.txt)
[CMakeOutput.log](https://github.com/Autodesk/maya-usd/files/5685969/CMakeOutput.log)
[CMakeError.log](https://github.com/Autodesk/maya-usd/files/5685970/CMakeError.log)

**Specs:**
 - OS: Windows 10
 - Compiler: Visual Studio 2017 
 - Maya 2020.2
 - Maya USD commit: v0.5.0 (62037f3874df7bd8393b4c4684b1d3952911af35) (also tried with v0.4.0)
 - Pixar USD commit: v20.11 (d8a405a1344480f859f025c4f97085143efacb53) (also tried with v20.02)
 - Python 2.7.15 64bit

**Additional context**

Repro for USD build:
- Open x64 Native Tools command prompt for VS 2017
- `git clone https://github.com/PixarAnimationStudios/USD.git`
- `git checkout v20.11`
- `python build_scripts\build_usd.py "C:\USD" --build-args boost,"--with-date_time --with-thread --with-system --with-filesystem"`

Repro for maya-usd build (**errors**):
- Open x64 Native Tools command prompt for VS 2017
- `git clone https://github.com/Autodesk/maya-usd.git`
- `git checkout v0.5.0`
- `python build.py --maya-location "C:\Program Files\Autodesk\Maya2020" --pxrusd-location C:\USD --devkit-location C:\devkitBase C:\workspace -v 3`

I tried using `--generator "Visual Studio 15 2017 Win64"`, or `--generator "Ninja"` (pointing args to ninja.exe), or using the `-DBUILD_STRICT_MODE=OFF`, but it looked like I had the same result in all cases.

Again, I would greatly appreciate any help that you could provide me. Cheers!  :)

**Comment by kxl-adsk** [comment]
Hi @abribanez,

All the instances of this issue in the past were related to the fact that one compiled USD without `--build-args boost,"--with-date_time --with-thread --with-system --with-filesystem"`.  In fact, this is exactly how I compile USD on my windows boxes for Maya 2020.

A few things for you to try:
1. Do a clean build of USD
2. Try building maya-usd with USD binaries from our installer (to confirm that issue is related to how USD is compiled)
3. Wait for our upcoming 0.6.0 binaries that will have USD 20.11

FWIW. I will convert this issue to discussion since for now there is nothing actionable for us.

Cheers,
KxL

**Reply by kxl-adsk** [reply]
Update -> 0.6.0 has been posted and has USD 20.11.

**Reply by abribanez** [reply]
:) awesome, yeah. I was able to install and succesfully loaded in Maya. Just still having issues with `usdview` as I mentioned in [here](https://github.com/Autodesk/maya-usd/discussions/766#discussioncomment-201651). This really is the reason why I'm trying to build maya-usd myself to get usdview up and running. Could this be related to the fact that I have python 2.7.15 64bit installed instead of python 3.x ?

**Reply by abribanez** [reply]
For now, I was able to solve the issue by using the USD version I built and adding the /bin and /lib to `PATH` and /lib/python to `PYTHONPATH`. It doesn't seem to affect/interfere with maya-usd, but I guess it would be cleaner to use `usdview` from the maya-usd install.

Regardless, thank you for taking the time to reply! Cheers!

**Comment by santosg87** [comment]
@abribanez the pySide2 issue is very likely to be due to the python version you have. one way to get around it is to use mayapy instead of python - since it does have PySide2 built.

assuming your PYTHONPATH is set properly, you should be able to call something like this:
`cd "C:\Program Files\Autodesk\Maya2020\bin"`
`mayapy "C:\Program Files\Autodesk\MayaUSD\Maya2020\0.6.0_202012102328-efea2cd\mayausd\USD\bin\usdview" [C:\temp.usd]`

where [C:\temp.usd] is your usd File to be opened. 

not the most elegant way - but it would at least get you using the usdview that comes packaged with 0.6.0 :)

**Reply by abribanez** [reply]
thats a good idea, thanks @santosg87! I'll give that a try for sure.

**Comment by seando-adsk** [comment]
As you pointed out, in order to run usdview, you need to have both the USD lib and bin folder in your path and the USD lib/python folder in your python path. You use your custom built USD or the one provided with the MayaUsd install.

You also need to have PyOpenGL installed into your python. So if you want to use mayapy, make sure you add PyOpenGL to it.
